x-pack/plugins/remote_clusters/public/sections/remote_cluster_list/components/disconnect_button/disconnect_button.js

```diff
@@ -40,48 +41,103 @@ export class DisconnectButtonUi extends Component {
+      <FormattedMessage
+        id="xpack.remoteClusters.disconnectButton.multipleButtonLabel"
+        defaultMessage="Disconnect {count} remote clusters"
+        values={{ count: clusterNames.length }}
+      />
+    );
+  }
+
   render() {
-    const { disconnectRemoteClusters, clusterNames, intl } = this.props;
+    const { intl, clusterNames } = this.props;
     const { isModalOpen } = this.state;
+    const isSingleCluster = clusterNames.length === 1;
 
     let modal;
 
     if (isModalOpen) {
+      const title = isSingleCluster ? intl.formatMessage({
+        id: 'xpack.remoteClusters.disconnectButton.confirmModal.deleteSingleClusterTitle',
+        defaultMessage: 'Disconnect remote cluster \'{name}\'?',
+      }, { name: clusterNames[0] }) : intl.formatMessage({
+        id: 'xpack.remoteClusters.disconnectButton.confirmModal.multipleDeletionTitle',
+        defaultMessage: 'Disconnect {count} remote clusters?',
+      }, { count: clusterNames.length });
+
+      const content = isSingleCluster ? null : (
+        <Fragment>
+          <p>
+            <FormattedMessage
+              id="xpack.remoteClusters.disconnectButton.confirmModal.multipleDeletionDescription"
+              defaultMessage="You are about to disconnect from {isSingleCluster, plural, one {this cluster} other {these clusters}}"
+              values={{ isSingleCluster: isSingleCluster ? 1 : 0 }}
+            />
+          </p>
+          {<ul>{clusterNames.map(name => <li key={name}>{name}</li>)}</ul>}
+        </Fragment>
+      );
+
       modal = (
         <EuiOverlayMask>
           <EuiConfirmModal
-            title={intl.formatMessage({
-              id: 'xpack.remoteClusters.disconnectButton.confirmModal.modalTitle',
-              defaultMessage: 'Disconnect remote clusters?',
-            })}
+            title={title}
             onCancel={this.closeConfirmModal}
-            onConfirm={() => disconnectRemoteClusters(clusterNames)}
+            onConfirm={this.onConfirm}
             cancelButtonText={
               intl.formatMessage({
                 id: 'xpack.remoteClusters.disconnectButton.confirmModal.cancelButtonText',
                 defaultMessage: 'Cancel',
               })
             }
+            buttonColor="danger"
             confirmButtonText={
               intl.formatMessage({
                 id: 'xpack.remoteClusters.disconnectButton.confirmModal.confirmButtonText',
                 defaultMessage: 'Disconnect',
               })
             }
-          />
+          >
+            {content}
+          </EuiConfirmModal>
         </EuiOverlayMask>
       );
     }
 
     return (
       <Fragment>
         <EuiButton color="danger" onClick={this.showConfirmModal}>
-          <FormattedMessage
-            id="xpack.remoteClusters.disconnectButton.buttonLabel"
-            defaultMessage="Disconnect remote clusters"
-          />
+          {this.renderButtonText()}
         </EuiButton>
-
         {modal}
```

Comment:
To me it would be cleaner to have a method to render the modal (same as `renderButtonText()`) and this line would be: `{ isModalOpen && this.renderModal() }`

Reply:
agreed, good future clean up item